### PR TITLE
scripts: tests: twister: Platform module test expansion

### PR DESCRIPTION
**Pull request by LukaszMrugala**
Added a unit test for the platform.py module.
It covers 99% of the code. The 1% is unreachable.

### CODE REVIEW

**Review by nashif — DISMISSED** [review]

File: scripts/tests/twister/test_platform.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 Intel Corporation
```
Comment (nashif) [inline scripts/tests/twister/test_platform.py:2]: 2023

**Review by nashif — APPROVED** [review]

**Comment by nashif** [comment]
@PerMac  @golowanow @gchwier please review

**Comment by golowanow** [comment]
@LukaszMrugala could you please rebase this PR to double-check for any regressions since 06/Dec/23

**Review by gchwier — APPROVED** [review]
rebased locally and tests still works

**Comment by LukaszMrugala** [comment]
> @LukaszMrugala could you please rebase this PR to double-check for any regressions since 13/Dec/23

Rebased locally, 2/2 tests pass.

**Review by PerMac — APPROVED** [review]

**Pull request merged**